components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java

```diff
@@ -1309,6 +1330,223 @@ public void addApplicationWithTemplateIdAndTemplateVersionData(String templateId
+
+        String resourceId = applicationManagementService.createApplication(inputSP, SUPER_TENANT_DOMAIN_NAME,
+                REGISTRY_SYSTEM_USERNAME);
+
+        // Retrieve added application with certificate.
+        when(applicationCertificateManagementService.getCertificate(anyInt(), anyString())).thenReturn(certificate);
+        ServiceProvider retrievedSP = applicationManagementService.getApplicationByResourceId(resourceId,
+                SUPER_TENANT_DOMAIN_NAME);
+        String certificateValue = Arrays.stream(retrievedSP.getSpProperties())
+                .filter(prop -> "CERTIFICATE".equals(prop.getName()))
+                .map(ServiceProviderProperty::getValue)
+                .findFirst()
+                .orElseThrow(() -> new AssertionError("Certificate property not found"));
+        Assert.assertEquals(Integer.parseInt(certificateValue), CERTIFICATE_ID);
+        Assert.assertEquals(retrievedSP.getCertificateContent(), CERTIFICATE);
+
+        // Update the application by updating the certificate.
+        doNothing().when(applicationCertificateManagementService).updateCertificateContent(anyInt(), anyString(),
+                anyString());
+        retrievedSP.setCertificateContent(UPDATED_CERTIFICATE);
+        try {
+            applicationManagementService.updateApplicationByResourceId(resourceId, retrievedSP,
+                    SUPER_TENANT_DOMAIN_NAME, REGISTRY_SYSTEM_USERNAME);
+        } catch (Exception e) {
+            Assert.fail();
+        }
+
+        // Update application by deleting certificate.
+        retrievedSP.setCertificateContent(StringUtils.EMPTY);
+        doNothing().when(applicationCertificateManagementService).updateCertificateContent(anyInt(), anyString(),
+                anyString());
+        try {
+            applicationManagementService.updateApplicationByResourceId(resourceId, retrievedSP,
+                    SUPER_TENANT_DOMAIN_NAME, REGISTRY_SYSTEM_USERNAME);
+        } catch (Exception e) {
+            Assert.fail();
+        }
+        ServiceProvider updatedSp = applicationManagementService.getApplicationByResourceId(resourceId,
+                SUPER_TENANT_DOMAIN_NAME);
+        certificateValue = Arrays.stream(retrievedSP.getSpProperties())
+                .filter(prop -> "CERTIFICATE".equals(prop.getName()))
+                .map(ServiceProviderProperty::getValue)
+                .findFirst()
+                .orElse(null);
+        Assert.assertNull(certificateValue);
+        Assert.assertNull(updatedSp.getCertificateContent());
+
+        // Deleting added application.
+        applicationManagementService.deleteApplication(inputSP.getApplicationName(), SUPER_TENANT_DOMAIN_NAME,
+                REGISTRY_SYSTEM_USERNAME);
+    }
+
+    @Test(dataProvider = "applicationCertificateDataProvider", dependsOnMethods = "testApplicationCertificateSuccess")
+    public void testApplicationCertificateClientError(ServiceProvider inputSP, Certificate certificate)
```

Comment:
Can we break to different methods

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6087/commits/9a2dc889d25ec1c2d6d950643a6310f1717fd44a